src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,545 @@
+
+        # Prepare the command
+        command = "jobs"
+        params = {"action": "status"}
+        query = self.base_url + command
+
+        # Submit the POST request to get status of the pilots
+        response = self.session.post(
+            query,
+            data=json.dumps(jobsJson),
+            headers=self.headers,
+            params=params,
+            timeout=self.arcRESTTimeout,
+        )
+        if not response.ok:
+            self.log.info("Failed getting the status of the jobs", "%s - %s" % (response.status_code, response.reason))
+            return S_ERROR("Failed getting the status of the jobs")
+
+        resultDict = {}
+        jobsToRenew = []
+        jobsToCancel = []
+        for job in response.json()["job"]:
+            jobID = self._arcToDiracID(job["id"])
+
+            # ARC REST interface returns hyperbole
+            arcState = job["state"].capitalize()
+            self.log.debug("REST ARC status", "for job %s is %s" % (jobID, arcState))
+            resultDict[jobID] = self.mapStates[arcState]
+
+            # Renew proxy only of jobs which are running or queuing
+            if arcState in ("Running", "Queuing"):
+                jobsToRenew.append(job["id"])
+            # Cancel held jobs so they don't sit in the queue forever
+            if arcState == "Hold":
+                jobsToCancel.append(job["id"])
+                self.log.debug("Killing held job %s" % jobID)
+
+        # Renew jobs to be renewed
+        # Does not work at present - wait for a new release of ARC CEs for this.
+        result = self._renewJobs(jobsToRenew)
+        if not result["OK"]:
+            return result
+
+        # Kill jobs to be killed
+        result = self.killJob(jobsToCancel)
+        if not result["OK"]:
+            return result
+
+        return S_OK(resultDict)
+
+    #############################################################################
+
+    def getJobOutput(self, jobID, localDir=None):
+        """Get the specified job standard output and error files.
+        The output is returned as strings.
+        """
+        if not self.session:
+            return S_ERROR("REST interface not initialised. Cannot get job output.")
+
+        # Get a proxy
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("AREXComputingElement: failed to set up proxy", result["Message"])
+            return result
+        self.session.cert = Locations.getProxyLocation()
+
+        # Extract stamp from the Job ID
+        if ":::" in jobID:
+            pilotRef, stamp = jobID.split(":::")
+        else:
+            pilotRef = jobID
+            stamp = ""
+        if not stamp:
+            return S_ERROR("Pilot stamp not defined for %s" % pilotRef)
+
+        # Prepare the command
+        command = "jobs/"
+        job = self._DiracToArcID(pilotRef)
+        query = self.base_url + command + job + "/session/" + stamp + ".out"
```

Comment:
I find these string manipulations inconvenient: there will always be the case when a double `/` or a double `.` is inserted. There are better libraries to avoid that, care to use them?

Reply:
I chose this specifically, as these strings are definite here. So, we any know exactly what we are doing and can avoid loading yet another library.

If it is a strong requirement, it can be done.

Reply:
> we any know exactly what we are doing
I like your positivity!

Loading another library is OK, anyway mine is not a strong requirement.

Reply:
Done - I had to write an internal function, as I did not find anything appropriate in the python libraries.